src/pip/_internal/network/auth.py

```diff
@@ -226,11 +230,13 @@ def _prompt_for_password(self, netloc):
 
     # Factored out to allow for easy patching in tests
     def _should_save_password_to_keyring(self):
+        # type: () -> bool
         if not keyring:
             return False
         return ask("Save credentials to keyring [y/N]: ", ["y", "n"]) == "y"
 
     def handle_401(self, resp, **kwargs):
+        # type: (Response, **Any) -> None
```

Comment:
Shouldn't the return type here be `Response`?

Reply:
Good catch, fixed in https://github.com/pypa/pip/pull/8277/commits/b5f82a0b34c2059083025431275b14a7032e3054

Reply:
Hmm... What does reveal_type(req) show, just before the final return? I'm curios why mypy didn't catch this. It might be that we should augument other code used here, which would be something we do in a follow up, but it'd be useful to take a look while this is fresh on our hands.

Reply:
@pradyunsg it's `Any`. Because `requests` has no typing?

Reply:
> Because `requests` has no typing?

I think these are the type annotations for requests? Requests does have type annotations (https://github.com/python/typeshed/tree/master/third_party/2and3/requests)